File: proto-java/src/test/java/com/spaceape/protobuf/reflect/FieldDescriptorTest.java
package com.spaceape.protobuf.reflect;

import com.spaceape.model.ProfileTO;
import org.testng.Assert;
import org.testng.annotations.Test;

import java.util.Map;

public class FieldDescriptorTest {
    @Test
    public void testAllFieldsInInheritance() throws Exception {
        ProfileTO profile = new ProfileTO();
        Map<Integer, FieldDescriptor> fields = profile.getFieldDescriptors();
        Assert.assertTrue(fields.containsKey(1));
    }
}

File: proto-csharp/ProtoMessage/ProtoMessage/Tests/com/spaceape/protobuf/reflect/FieldDescriptorTest.cs
using System;
using System.IO;
using System.Collections.Generic;
using NUnit.Framework;
using com.spaceape.protobuf;
using com.spaceape.common;
using com.spaceape.model2;
using com.spaceape.model;
using Google.ProtocolBuffers;

namespace com.spaceape.protobuf.reflect
{
  [TestFixture()]
  public class FieldDescriptorTest
  {
    [Test()]
    public void testAllFieldsInInheritance()
    {
      ProfileTO profile = new ProfileTO();
      Dictionary<int, FieldDescriptor> fields = profile.getFieldDescriptors();
      Assert.IsTrue(fields.ContainsKey(1));
    }
  }
}

